src/main/java/org/takes/misc/Concat.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.misc;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Concat iterable.
+ *
+ * @author Jason Wong (super132j@yahoo.com)
+ * @version $Id$
+ * @since 0.13.8
+ */
+public final class Concat<T> implements Iterable<T> {
+
+    /**
+     * Internal storage to hold the elements from iterables.
+     */
+    private final transient List<T> storage = new LinkedList<T>();
```

Comment:
@krzyk, I have 2 questions regarding to holding references:

1. How are we implement ```iterator()``` method? Do we need to write a brand new iterator which calls the iterators of the two iterables?
2. How are the use cases of ```Concat```? If there are more use cases that the input iterable will eventually be garbage collected then I think creating one new object would save more memory. I think when we use ```Concat```, we would be more interested in the end result rather than the input and the input would then be unreferenced. 
3. How would we handle ```remove()``` method that could be implemented by iterables? If the input iterable is modified by this method then it would affect the concatenated result if we rely on the reference to produce the output.

Thanks.